File: demo_java/src/main/java/com/techcentrix/sdk/demo_java/ui/SettingsFragment.java
package com.techcentrix.sdk.demo_java.ui;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.preference.ListPreference;
import androidx.preference.PreferenceFragmentCompat;

import com.techcentrix.sdk.demo_java.R;
import com.techcentrix.sdk.demo_java.util.ThemeHelper;

public class SettingsFragment extends PreferenceFragmentCompat {

    static final String TAG = "SettingsFragment";

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        setPreferencesFromResource(R.xml.settings, rootKey);

        @Nullable ListPreference themePreference = findPreference(ThemeHelper.PREFS_KEY);
        if (themePreference != null) {
            themePreference.setOnPreferenceChangeListener((preference, newValue) -> {
                ThemeHelper.applyThemeFromPreferenceValue((String) newValue);
                return true;
            });
        }
    }
}

File: demo_kotlin/src/main/java/com/techcentrix/sdk/demo_kotlin/ui/SettingsFragment.kt
package com.techcentrix.sdk.demo_kotlin.ui

import android.os.Bundle
import androidx.preference.ListPreference
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import com.techcentrix.sdk.demo_kotlin.R
import com.techcentrix.sdk.demo_kotlin.util.ThemeHelper

class SettingsFragment : PreferenceFragmentCompat() {

    companion object {
        const val TAG = "SettingsFragment"
    }

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        setPreferencesFromResource(R.xml.settings, rootKey)

        val themePreference: ListPreference = requireNotNull(findPreference(ThemeHelper.PREFS_KEY))
        themePreference.onPreferenceChangeListener =
            Preference.OnPreferenceChangeListener { _, newValue ->
                ThemeHelper.applyThemeFromPreferenceValue(newValue as String)
                true
            }
    }
}
